x-pack/plugins/observability/public/hooks/slo/use_fetch_apm_indices.ts

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { HttpSetup } from '@kbn/core-http-browser';
+import { useCallback } from 'react';
+import { useDataFetcher } from '../use_data_fetcher';
+
+type ApmIndex = string;
+
+export interface UseFetchApmIndex {
+  data: ApmIndex;
+  loading: boolean;
+  error: boolean;
+}
+
+interface ApiResponse {
+  apmIndexSettings: Array<{
+    configurationName: string;
+    defaultValue: string;
+    savedValue?: string;
+  }>;
+}
+
+export function useFetchApmIndex(): UseFetchApmIndex {
```

Comment:
You could port this to React Query, but that can also happen later; I'll leave that judgement to you.